control-admin/src/main/java/fi/nls/oskari/control/admin/GetAllRolesHandler.java

```diff
@@ -33,7 +38,14 @@ public void init() {
     }
 
     @Override
-    public void handleAction(ActionParameters params) throws ActionException {
+    public void preProcess(ActionParameters params) throws ActionException {
+        if (!params.getUser().isAdmin()) {
```

Comment:
Could have used ActionParameters#requireAdminUser();

Also could move the userService == null check from handleGet() to preProcess()

Reply:
Updated PR as suggested.